server/src/main/java/org/elasticsearch/action/admin/indices/settings/get/TransportGetSettingsAction.java

```diff
@@ -73,21 +77,27 @@ protected GetSettingsResponse newResponse() {
     protected void masterOperation(GetSettingsRequest request, ClusterState state, ActionListener<GetSettingsResponse> listener) {
         Index[] concreteIndices = indexNameExpressionResolver.concreteIndices(state, request);
         ImmutableOpenMap.Builder<String, Settings> indexToSettingsBuilder = ImmutableOpenMap.builder();
+        ImmutableOpenMap.Builder<String, Settings> indexToDefaultSettingsBuilder = ImmutableOpenMap.builder();
         for (Index concreteIndex : concreteIndices) {
             IndexMetaData indexMetaData = state.getMetaData().index(concreteIndex);
             if (indexMetaData == null) {
                 continue;
             }
 
-            Settings settings = settingsFilter.filter(indexMetaData.getSettings());
+            Settings indexSettings = settingsFilter.filter(indexMetaData.getSettings());
             if (request.humanReadable()) {
-                settings = IndexMetaData.addHumanReadableSettings(settings);
+                indexSettings = IndexMetaData.addHumanReadableSettings(this.settings);
             }
+
             if (CollectionUtils.isEmpty(request.names()) == false) {
-                settings = settings.filter(k -> Regex.simpleMatch(request.names(), k));
+                indexSettings = indexSettings.filter(k -> Regex.simpleMatch(request.names(), k));
+            }
+
+            indexToSettingsBuilder.put(concreteIndex.getName(), indexSettings);
+            if (request.includeDefaults()) {
+                indexToDefaultSettingsBuilder.put(concreteIndex.getName(), indexScopedSettings.diff(indexSettings, this.settings));
```

Comment:
don't we also need to call settingsFilter.filter on the result of the diff?

Reply:
Perhaps!  I assumed that the this.settings object would not contain anything that needed to be filtered, but I might be incorrect -- do you have any insights on how settingsFilter and this.settings are created?

Reply:
I am not sure either, I would need to test this, I only saw that we were using it before in the REST action, but I wonder too if it was needed there.